.github/workflows/firebase-hosting-preview.yml

```diff
@@ -41,21 +41,82 @@ jobs:
         with:
           node-version: 'lts/*'
 
-      - name: Install dependencies and build
-        run: |
-          npm ci
-          npm run build
+      - name: Install dependencies
+        run: npm ci
+
+      # Inject the App Check Debug Token into the .env file at build time
+      - name: Create .env file with App Check Debug Token
+        run: echo "VITE_APPCHECK_DEBUG_TOKEN=${{ secrets.VITE_APPCHECK_DEBUG_TOKEN }}" >> .env
+
+      - name: Build project
+        run: npm run build
 
       - name: Deploy to Firebase Hosting Channel
         id: firebase-deploy
         uses: FirebaseExtended/action-hosting-deploy@v0
         with:
-          repoToken: '${{ secrets.GITHUB_TOKEN }}'
-          firebaseServiceAccount: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_GSE_ROAR_ADMIN }}'
+          repoToken: ${{ secrets.GITHUB_TOKEN }}
+          firebaseServiceAccount: ${{ secrets.FIREBASE_SERVICE_ACCOUNT_GSE_ROAR_ADMIN }}
           projectId: gse-roar-admin
           target: staging
 
-      - run: echo ${{ fromJson(steps.firebase-deploy.outputs.urls)[0] }}
+      - name: Set Output Deploy URL
+        id: set-deploy-url
+        run: echo "deployUrl=${{ fromJson(steps.firebase-deploy.outputs.urls)[0] }}" >> $GITHUB_OUTPUT
 
     outputs:
-      deployUrl: ${{ fromJson(steps.firebase-deploy.outputs.urls)[0] }}
+      deployUrl: ${{ steps.set-deploy-url.outputs.deployUrl }}
+
+  whitelist-pr-domains-roar-admin:
+    name: Whitelist PR Domains on Firebase App Check for gse-roar-admin
+    runs-on: ubuntu-latest
+    needs: build-and-preview
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Authenticate with Google Cloud SDK
+        uses: google-github-actions/auth@v2
+        with:
+          credentials_json: ${{ secrets.FIREBASE_SERVICE_ACCOUNT_GSE_ROAR_ADMIN }}
+
+      - name: Set up Google Cloud SDK with Firebase Service Account
+        uses: google-github-actions/setup-gcloud@v2
+        with:
+          project_id: gse-roar-admin
+
+      - name: Update App Check Domains for gse-roar-admin
+        run: |
+          CURRENT_DOMAINS=$(gcloud recaptcha keys describe ${{ secrets.FIREBASE_APPCHECK_SITE_KEY_GSE_ROAR_ADMIN }} --format="value(webSettings.allowedDomains)" | tr ';' '\n')
+          NEW_DOMAIN=$(echo ${{ needs.build-and-preview.outputs.deployUrl }} | sed 's|https://||')
+          ALL_DOMAINS=$(echo -e "${CURRENT_DOMAINS}\n${NEW_DOMAIN}")
+          UPDATED_DOMAINS=$(echo "$ALL_DOMAINS" | sort | uniq | paste -sd ',' -)
+          gcloud recaptcha keys update ${{ secrets.FIREBASE_APPCHECK_SITE_KEY_GSE_ROAR_ADMIN }} --domains="$UPDATED_DOMAINS" --web
```

Comment:
Not a requirement per-se but this would be a great candidate for a reusable action to avoid the duplication between `whitelist-pr-domains-roar-admin` and `whitelist-pr-domains-roar-assessment`

Reply:
I also thought about making the whitelisting job its own workflow, but I was not sure how to capture the deployment link output from the Build and Deploy job and use it in a different workflow. If that's even possible?

Reply:
This should be possible by extending the existing workflows to create deployments and register environments. For each PR environment, the workflow would automatically create a new environment with the corresponding PR URL. 

The separate workflow would listen to `deployment` events and then identify the URL from that deployment. 

Of course this would need to be extended further to deactivate environments and subsequently remove the the URL from the whitelist